### PR TITLE
Improve components

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes
| BC breaks?    | maybe :D
| Deprecations? | no
| Tests pass?   | yes
| Tickets | #1769 #1766 #1764 #1765 #1768

### CODE REVIEW

File: inc/devicegeneric.class.php [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      $column = parent::getHTMLTableHeader($itemtype, $base, $super, $father, $options);
+
+      if ($column == $father) {
+         return $father;
+      }
+
+      switch ($itemtype) {
+         case 'Computer' :
+            Manufacturer::getHTMLTableHeader(__CLASS__, $base, $super, $father, $options);
+            //InterfaceType::getHTMLTableHeader(__CLASS__, $base, $super, $father, $options);
```
Comment (orthagh) [inline inc/devicegeneric.class.php]: comments ?

File: inc/devicegeneric.class.php [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+      if ($column == $father) {
+         return $father;
+      }
+
+      switch ($item->getType()) {
+         case 'Computer' :
+            Manufacturer::getHTMLTableCellsForItem($row, $this, NULL, $options);
+            //InterfaceType::getHTMLTableCellsForItem($row, $this, NULL, $options);
+            /*
```
Comment (orthagh) [inline inc/devicegeneric.class.php]: comments ?

File: inc/deviceharddrivemodel.class.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+/// Class DeviceHardDriveModel
+class DeviceHardDriveModel extends CommonDropdown {
+
+   var $additional_fields_for_dictionnary = array('manufacturer');
+
+
+   static function getTypeName($nb=0) {
+      return _n('Device harddrive model', 'Device harddrive models', $nb);
```
Comment (orthagh) [inline inc/deviceharddrivemodel.class.php]: maybe a space -> hard drive

File: inc/devicemotherboardmodel.class.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+/// Class DeviceMotherBoardModel
+class DeviceMotherBoardModel extends CommonDropdown {
+
+   var $additional_fields_for_dictionnary = array('manufacturer');
+
+
+   static function getTypeName($nb=0) {
+      return _n('Device motherboard model', 'Device motherboard models', $nb);
```
Comment (orthagh) [inline inc/devicemotherboardmodel.class.php]: DeviceMotherboard label is 'System board'

File: inc/devicenetworkcardmodel.class.php [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+/// Class DeviceNetworkCardModel
+class DeviceNetworkCardModel extends CommonDropdown {
+
+   var $additional_fields_for_dictionnary = array('manufacturer');
+
+
+   static function getTypeName($nb=0) {
+      return _n('Device networkcard model', 'Device networkcard models', $nb);
```
Comment (orthagh) [inline inc/devicenetworkcardmodel.class.php]: 'Network card' for the DeviceNetworkCard

File: inc/devicepcimodel.class.php [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+/// Class DevicePciModel
+class DevicePciModel extends CommonDropdown {
+
+   var $additional_fields_for_dictionnary = array('manufacturer');
+
+
+   static function getTypeName($nb=0) {
+      return _n('Device hard drive model', 'Device hard drive models', $nb);
```
Comment (orthagh) [inline inc/devicepcimodel.class.php]: DevicePci have 'Other component', but anyway hard drive is bad :)

File: inc/devicesimcard.class.php [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+                                     'type'  => 'integer')
+                           ));
+   }
+   */
+
+   function getSearchOptionsNew() {
+
+      $tab                 = parent::getSearchOptionsNew();
+
+      $tab[] = [
```
Comment (orthagh) [inline inc/devicesimcard.class.php]: i don't get the point of theses searchoptions for a simcard.
frequency, nb cores ?

File: inc/item_devices.class.php [hunk 8]
```diff
@@ -169,15 +200,24 @@ static function getSpecificities($specif='') {
    /**
     * Get the items on which this Item_Device can be attached. For instance, a computer can have
     * any kind of device. Conversely, a soundcard does not concern a NetworkEquipment
+    * A configuration entry is automatically checked in $CFG_GLPI (must be the name of
+    * the calss, lowercase, without "_" with extra "_types" at the end; for example
```
Comment (orthagh) [inline inc/item_devices.class.php]: typo: calss -> class

File: inc/item_devices.class.php [hunk 9]
```diff
@@ -169,15 +200,24 @@ static function getSpecificities($specif='') {
...
+    * "itemdevicesoundcard_types").
     *
-    * Should be overloaded by Item_Device*
+    * Alternatvely, it could be overloaded from subclasses
```
Comment (orthagh) [inline inc/item_devices.class.php]: typo: Alternatively

File: inc/item_devices.class.php [hunk 10]
```diff
@@ -1007,4 +1070,44 @@ function showForm($ID, $options=array()) {
...
+   function prepareInputForAdd($input) {
+      global $DB, $CFG_GLPI;
+
+      $computer = static::getItemFromArray(static::$itemtype_1, static::$items_id_1, $input);
+
+      if ($CFG_GLPI["is_location_autoupdate"]
+         && ($computer->fields['locations_id'] != $input['locations_id'])
+         ||!isset($input['locations_id'])
+      ) {
+         $input['locations_id'] = addslashes($computer->fields['locations_id']);
```
Comment (orthagh) [inline inc/item_devices.class.php]: addslashes ?

File: inc/item_devicesimcard.class.php [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+   static public $items_id_2 = 'devicesimcards_id';
+
+   static protected $notable = false;
+
+
+   /**
+    * @since version 0.85
+    **/
+   /*
+   static function getSpecificities($specif='') {
```
Comment (orthagh) [inline inc/item_devicesimcard.class.php]: comments  and processors specificities ?

File: inc/computer.class.php [hunk 12]
```diff
@@ -289,6 +288,34 @@ function post_updateItem($history=1) {
...
+
+         //fields that are not present for devices
+         unset($changes['groups_id']);
+         unset($changes['users_id']);
+         unset($changes['contact_num']);
+         unset($changes['contact']);
+
+         if (count($changes) > 0) {
+            // Propagates the changes to linked devices
+            foreach ($CFG_GLPI['itemdevices'] as $t) {
```
Comment (trasher) [inline inc/computer.class.php]: Since it was already "$t" I did not change it.

File: inc/devicepcimodel.class.php (hunk 6 above)
Comment (trasher) [inline inc/devicepcimodel.class.php]: Yes, but other component is not enough generic apparently.

File: inc/devicesimcard.class.php (hunk 7 above)
Comment (trasher) [inline inc/devicesimcard.class.php]: Will drop this one; this is not ok.

File: inc/item_devicesimcard.class.php (hunk 11 above)
Comment (trasher) [inline inc/item_devicesimcard.class.php]: Will be dropped

**Pull request merged**